Scrollby markdown (#6138)
Codecademy/docs: content/javascript/concepts/window/terms/scrollBy/scrollBy.md

---

> Title: 'scrollBy()'
> Description: 'Scrolls the window's content by a specified number of pixels, relative to the current scroll position.'
> Subjects:
>   - 'Web Development'
>   - 'Computer Science'
> Tags:
>   - 'Functions'
>   - 'Parameters'
>   - 'Arguments'
> CatalogContent:
>   - 'introduction-to-javascript'
>   - 'paths/front-end-engineer-career-path'

Reviewer: ```suggestion
Title: '.scrollBy()'
Description: 'Scrolls the window content by a specified number of pixels, relative to the current scroll position'
Subjects:
  - 'Computer Science'
  - 'Web Development'
Tags:
  - 'Arguments'
  - 'Functions'
  - 'Parameters'
CatalogContent:
  - 'introduction-to-javascript'
  - 'paths/front-end-engineer-career-path'
```

Reviewer: tags and subjects should be in alphabetical order. 
Avoid apostrophe in description

---

> The **`scrollBy()`** method is a function provided by the browser's window object. The method will scroll the user's window down, right, left, or up, depending on the parameters input into the method. The scrolling occurs relative to the current scroll position, meaning that if the user has already scrolled 100 pixels down a page, using `scrollBy()` to scroll a further 100 pixels down will cause the user to be 200 pixels down the page.

Reviewer: ```suggestion
The **`.scrollBy()`** method is a function provided by the browser's [window](https://www.codecademy.com/resources/docs/javascript/window) object. It scrolls the window content up, down, left, or right based on the specified parameters. The scrolling occurs relative to the current scroll position, meaning if the user has already scrolled 100 pixels down and `scrollBy(0, 100)` is called, the new scroll position will be 200 pixels down the page.
```

Reviewer: added link for window object, reframed a bit

## Syntax

`scrollBy()` receives up to two parameters. The first is a number, which represents the number of pixels the user desires to scroll horizontally, for the x-coordinate. The second is a number, which represents the number of pixels the user desires to scroll vertically, for the y-coordinate.

These numbers can be **positive** or **negative**. Positive numbers in the x-coordinate parameter will scroll right, and positive numbers in the y-coordinate parameter will scroll down. Negative numbers in the x-coordinate parameter will scroll left, and negative numbers in the y-coordinate parameter will scroll up.

> Below are some examples of how `scrollBy()` is composed:

Reviewer: these need to be added below the syntax as:

- `x-coordinate`: <description>
- `y-coordinate`: <description>

Putting them in bullets would make the doc more readable

```pseudo

window.scrollBy(x-coordinate,y-coordinate);
```

> 

Reviewer: here, you can also add the alternative syntax as:

Alternatively, there is another syntax as follows:
```pseudo
scrollBy(options)
```

- `options`: <description>

## Example 1

> The following example uses the `scrollBy()` function to scroll a window down when a button is pressed.

Reviewer: ```suggestion
The following example uses the `scrollBy()` function to scroll the window down by _200_ pixels when a button is clicked:
```

```js

> // Creating a reference to a scroll button on an external HTML document.
> const scrollButton = document.getElementById("scrollButton");
> 
> // When the button is pressed, the window scrolls down by 200 pixels.
> scrollButton.addEventListener("click", function() {
> 	window.scrollBy(0,200); 
> });

Reviewer: ```suggestion
// Get a reference to the button element.
const scrollButton = document.getElementById("scrollButton");

// Scroll down by 200 pixels when the button is clicked.
scrollButton.addEventListener("click", function() {
  window.scrollBy(0, 200); 
});
```

Reviewer: reframed comments

```
## Example 2

> The following example uses the `scrollBy()` function to scroll a window to the left and up 100 pixels each. This syntax demonstrates using `scrollBy()` with negative numbers.

Reviewer: ```suggestion
The following example scrolls the window _100_ pixels to the left and _100_ pixels up using negative values:
```

```js

> // Creating a reference to a scroll button on an external HTML document.
> const scrollButton2 = document.getElementById("scrollButton2");
> 
> // When the button is pressed, the window scrolls left by 100 pixels and up by 100 pixels.
> scrollButton2.addEventListener("click", function() {
> 	window.scrollBy(-100,-100);

Reviewer: ```suggestion
// Get a reference to the second button element.
const scrollButton2 = document.getElementById("scrollButton2");

// Scroll left by 100 pixels and up by 100 pixels when the button is clicked.
scrollButton2.addEventListener("click", function() {
  window.scrollBy(-100, -100);
});

```

Reviewer: reframed comments and fixed code indentation (it should be 2 spaces only)
